### ISSUE TITLE
Hashes in extconfig

### ISSUE THREAD

**Issue by mat1010** [oxc/puppet-dovecot#19]
Is there a way adding hashes into an `extconfig`? Most of my extconfigs follow the rules of the module and it works great so far. But I have one extconfig, for acls that are saved in the database, that would require a hash.

The final extconfig should look like this:
```
connect = host=127.0.0.1 dbname=mail user=username password=password
map {
  pattern = shared/shared-boxes/user/$to/$from
  table = user_shares
  value_field = dummy
  fields {
    from_user = $from
    to_user = $to
  }
}
map {
  pattern = shared/shared-boxes/anyone/$from
  table = anyone_shares
  value_field = dummy

  fields {
    from_user = $from
  }
}
```
The extconfigs resource I tried it with looks like this
```
dovecot::extconfigs:
  'dovecot-dict-sql.conf.ext':
    connect: "host=127.0.0.1 dbname=mail user=imap%{serverid} password=%{lookup('imap_mysql_password')}"
    map:
      pattern: shared/shared-boxes/user/$to/$from
      table: user_shares
      value_field: dummy
      fields:
        from_user: $from
        to_user: $to
    map:
      pattern: shared/shared-boxes/anyone/$from
      table: anyone_shares
      value_field: dummy
      fields:
        from_user: $from
```

Right now this fails, since only strings are allowed in the key: value mappings. Is there a way arround it?

Thanks in advance

**Comment by oxc** [comment]
I didn't know that was a thing :) Can you provide a link to the documentation of that configuration?

**Comment by mat1010** [comment]
Wow - that was a quick response :-D

Here you go: https://wiki.dovecot.org/SharedMailboxes/Shared it's mentioned in the section "Using SQL dictionary"

**Comment by oxc** [comment]
Wow, that's crazy. Since this is a very special syntax, I'm inclined to support this only via a extra_content property in the extfile where you can put verbatim content (probably using yaml block notatoin). Would this be sufficient for your use case?

**Comment by mat1010** [comment]
That would be completly ok for me. I just want to avoid having to define file resources that add configs into the dovecot config directory which are unknown / not managed by the dovecot module itself.

**Comment by oxc** [comment]
@mat1010, can you please test the proposed fix in #20?

You can find a syntax example in the (updated) README

**Comment by mat1010** [comment]
Yes - it works like expected. Thank you so much for that fast response and providing a fix!

**Comment by oxc** [comment]
Released on the forge as v2.2.0 (with a release date of tomorrow :rofl: got the 20s mixed up...)
